torch/_dynamo/variables/torch_function.py

```diff
@@ -597,8 +597,9 @@ def from_tensor_var(cls, tx, tensor_var, class_type, cls_source):
 
         # This simulates shallow-copying the tensor object.
         kwargs = dict(tensor_var.__dict__)
-        assert kwargs.pop("class_type") is torch.Tensor, (
-            "invalid class type in TensorWithTFOverrideVariable.from_tensor_var"
+        input_tensor_type = kwargs.pop("class_type")
+        assert input_tensor_type in (torch.Tensor, torch.nn.Parameter), (
```

Comment:
This happens if an input tensor is `Parameter(TensorSubclass(...))`, and we call `.to(same_device)`, because the example_value won't change and would preserve the Parameter-ness into `class_type`.